### PR TITLE
WebSocket RPC + GUI in a separate process

### PR DESCRIPTION
**Pull request by mfranciszkiewicz**
Expected future improvements on RPC:
- reducing the size and complexity of exchanged objects
- choosing of a more lightweight and platform-independent serialization method (e.g. common to web apps)
- keyring / key auth implementation (basic support included)

GUI changes proposed:
- preview updates should be performed in a separate process

Also, a better process monitor might be required.

### CODE REVIEW

**Comment by chfast** [comment]
Only if you know how to do it, you could squeeze some of the WIP commits to make the history look better.

**Comment by badb** [comment]
We must change `start task` method just after this update, start task just after add (and remove start task button and context menu method) or made button and method not enabled just after click.

File: gnr/customizers/configurationdialogcustomizer.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 from PyQt4 import QtCore
 from PyQt4.QtGui import QMessageBox, QPalette
+from twisted.internet.defer import inlineCallbacks
```
Comment (badb) [inline gnr/customizers/configurationdialogcustomizer.py]: This isn't used.

File: gnr/gnrapplicationlogic.py [hunk 2]
```diff
@@ -231,9 +293,10 @@ def _get_builder(self, task_state):
         if hasattr(task_state.definition, "renderer"):
             task_state.definition.task_type = task_state.definition.renderer
 
-        return self.task_types[task_state.definition.task_type].task_builder_type(self.client.get_node_name(),
-                                                                                  task_state.definition,
-                                                                                  self.client.datadir)
+        builder = self.task_types[task_state.definition.task_type].task_builder_type(self.node_name,
```
Comment (badb) [inline gnr/gnrapplicationlogic.py:296]: We would like to have "change_node_name" function in the future available and `node_name` is set only at the beginning. It's not really important right now, just a look-out for the future.

File: golem/client.py [hunk 3]
```diff
@@ -162,21 +178,27 @@ def start_network(self):
         self.task_server.task_manager.register_listener(ClientTaskManagerEventListener(self))
         self.p2pservice.connect_to_network()
 
+        self.do_work_task.start(0.1, False)
```
Comment (badb) [inline golem/client.py]: Can you move it to `start` function?

File: tests/golem/core/test_processmonitor.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+        if all_stopped:
+            break
+        else:
+            time.sleep(0.5)
+
+
+class TestProcessMonitor(unittest.TestCase):
+
+    def test_monitor(self):
```
Comment (badb) [inline tests/golem/core/test_processmonitor.py:50]: On windows I get an error: 

> PicklingError: Can't pickle <function run_exit at 0x02CBDFB0>: it's not found as test_processmonitor.run_exit

File: gnr/gnrapplicationlogic.py (hunk 2 above)
Comment (mfranciszkiewicz) [inline gnr/gnrapplicationlogic.py:296]: Will `change_node_name` be a logic's method or a part of `change_config`? If so, there will be no problem keeping it up to date. Inserting a yield / deferred call here complicates testing.

Comment (badb) [inline gnr/gnrapplicationlogic.py:296]: Part of `change_config` probably, but I'm not sure yet. Anyway, it's all right, I've just added a node to #438 to remember about this.

**Pull request merged**